### PR TITLE
[dag][forge] dag onchain enable test

### PR DESCRIPTION
**Pull request by ibalajiarun**
### Description

Adds a Forge test to test onchain config change to enable DAG and makes adds DAG-specific QS configs.

### CODE REVIEW

**Comment by ibalajiarun** [comment]
Current dependencies on/for this PR:
* main
  * **PR #10953** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10953?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #10799** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10799?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10953?utm_source=stack-comment).

File: config/src/config/quorum_store_config.rs [hunk 1]
```diff
@@ -107,6 +107,29 @@ impl Default for QuorumStoreConfig {
 }
 
 impl QuorumStoreConfig {
+    pub fn default_for_dag() -> Self {
+        Self {
+            sender_max_batch_txns: 300,
```
Comment (zekun000) [inline config/src/config/quorum_store_config.rs:112]: probably leave a comment on the difference between the two default

File: testsuite/testcases/src/dag_onchain_enable_test.rs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        let mut cli = runtime.block_on(async {
+            CliTestFramework::new(
+                swarm.validators().next().unwrap().rest_api_endpoint(),
+                faucet_endpoint,
+                /*num_cli_accounts=*/ 0,
+            )
+            .await
+        });
+
+        std::thread::sleep(duration / 2);
```
Comment (zekun000) [inline testsuite/testcases/src/dag_onchain_enable_test.rs:48]: why?

File: testsuite/testcases/src/dag_onchain_enable_test.rs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    }
+}
+
+impl NetworkTest for DagOnChainEnableTest {
+    fn run(&self, ctx: &mut aptos_forge::NetworkContext<'_>) -> anyhow::Result<()> {
+        <dyn NetworkLoadTest>::run(self, ctx)
+    }
+}
+
+fn generate_blob(data: &[u8]) -> String {
```
Comment (zekun000) [inline testsuite/testcases/src/dag_onchain_enable_test.rs:113]: duplicate function?

**Review by zekun000 — APPROVED** [review]

File: testsuite/testcases/src/dag_onchain_enable_test.rs (hunk 2 above)
Comment (ibalajiarun) [reply testsuite/testcases/src/dag_onchain_enable_test.rs:48]: we let the current consensus run with load for some time before we flip the flag.

**Review by sasha8 — APPROVED** [review]

**Pull request merged**